kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/formats/AvroFormatWriter.scala

```diff
@@ -49,12 +47,10 @@ class AvroFormatWriter(outputStreamFn: () => S3OutputStream) extends S3FormatWri
 
   }
 
-  override def close(): Unit = {
-    avroWriterState.fold(logger.debug("Requesting close when there's nothing to close"))(_.close())
+  override def close(newName: BucketAndPath): Unit = {
```

Comment:
Is there a reason why this has to be passed to the close function? Wouldn't this information be part of the writer internal state ?

Reply:
The term "writer" applies in the S3 Sink to multiple writers.

We have the S3 Writer and the Format Writer.
The S3 writer has the internal state including topics/partitions etc.
The format writer knows nothing of the internal state and is just handling streams.

As we need to give it a new file name on close then it seems to fit in best here.  It means I was able to get rid of the logic between layers where I was keeping track of whether we have an outstanding rename in the S3 writer after coming to the end of the stream at the Format writer.

It seems to simplify things, anyway.